--- rules.orig	2016-07-24 21:11:49.000000000 +0800
+++ rules	2016-07-25 03:47:13.000000000 +0800
@@ -52,9 +52,11 @@
 	--with-mail_ssl_module \
 	--with-file-aio \
 	--with-ipv6 \
+	--with-openssl=$(SSLPACK) \
 	$(WITH_HTTP2) \
 	--with-cc-opt="$(CFLAGS)" \
-	--with-ld-opt="$(LDFLAGS)"
+	--with-ld-opt="$(LDFLAGS)" \
+	$(EXTRACONFIG)
 
 %:
 	dh $@ 
@@ -65,6 +67,7 @@
 	dh_strip --dbg-package=nginx-dbg
 
 override_dh_auto_build:
+	touch $(SSLPACK)/.openssl/include/openssl/ssl.h || true
 	dh_auto_build
 	mv objs/nginx objs/nginx-debug
 	mv objs/ngx_http_xslt_filter_module.so objs/ngx_http_xslt_filter_module-debug.so
@@ -74,6 +77,7 @@
 	mv objs/src/http/modules/perl/blib/arch/auto/nginx/nginx.so objs/src/http/modules/perl/blib/arch/auto/nginx/nginx-debug.so
 	mv objs/ngx_http_js_module.so objs/ngx_http_js_module-debug.so
 	CFLAGS="" ./configure $(COMMON_CONFIGURE_ARGS)
+	touch $(SSLPACK)/.openssl/include/openssl/ssl.h || true
 	dh_auto_build
 
 configure_debug:
