--- nginx.spec.orig	2016-07-23 16:53:03.000000000 +0800
+++ nginx.spec	2016-07-23 23:35:47.000000000 +0800
@@ -13,7 +13,6 @@
 Requires: initscripts >= 8.36
 Requires(post): chkconfig
 Requires: openssl
-BuildRequires: openssl-devel
 BuildRequires: perl
 BuildRequires: GeoIP-devel
 %endif
@@ -25,7 +24,6 @@
 Requires: initscripts >= 8.36
 Requires(post): chkconfig
 Requires: openssl >= 1.0.1
-BuildRequires: openssl-devel >= 1.0.1
 BuildRequires: perl-devel
 BuildRequires: perl-ExtUtils-Embed
 BuildRequires: GeoIP-devel
@@ -40,7 +38,6 @@
 Requires: systemd
 Requires: openssl >= 1.0.1
 BuildRequires: systemd
-BuildRequires: openssl-devel >= 1.0.1
 BuildRequires: perl-devel
 BuildRequires: perl-ExtUtils-Embed
 BuildRequires: GeoIP-devel
@@ -120,6 +117,8 @@
         --with-mail_ssl_module \
         --with-file-aio \
         --with-ipv6 \
+        --with-openssl=%{SSLPACK} \
+        %{?with_http2:--with-http_spdy_module} \
         %{?with_http2:--with-http_v2_module}")
 
 Summary: High performance web server
@@ -143,6 +142,7 @@
 Source11: nginx-debug.service
 Source12: COPYRIGHT
 Source13: njs-%{module_njs_shaid}.tar.gz
+Patch0: %{SPDY_PATCH}
 
 License: 2-clause BSD-like license
 
@@ -211,6 +211,8 @@
 %prep
 %setup -q
 tar xvzf %SOURCE13
+%patch0 -p1
+mv %_sourcedir/%{SSLPACK} .
 cp %{SOURCE2} .
 sed -e 's|%%DEFAULTSTART%%|2 3 4 5|g' -e 's|%%DEFAULTSTOP%%|0 1 6|g' \
     -e 's|%%PROVIDES%%|nginx|g' < %{SOURCE2} > nginx.init
@@ -220,7 +222,9 @@
 %build
 ./configure %{COMMON_CONFIGURE_ARGS} \
     --with-cc-opt="%{WITH_CC_OPT}" \
-    --with-debug
+    --with-debug \
+    %{EXTRACONFIG}
+touch $(SSLPACK)/.openssl/include/openssl/ssl.h || true
 make %{?_smp_mflags}
 %{__mv} %{bdir}/objs/nginx \
     %{bdir}/objs/nginx-debug
@@ -237,7 +241,9 @@
 %{__mv} %{bdir}/objs/ngx_http_js_module.so \
     %{bdir}/objs/ngx_http_js_module-debug.so
 ./configure %{COMMON_CONFIGURE_ARGS} \
-    --with-cc-opt="%{WITH_CC_OPT}"
+    --with-cc-opt="%{WITH_CC_OPT}" \
+    %{EXTRACONFIG}
+touch $(SSLPACK)/.openssl/include/openssl/ssl.h || true
 make %{?_smp_mflags}
 
 %install
